rfcs/0006-implementation-api.md

```diff
@@ -0,0 +1,47 @@
+# Storage API
+
+**suRaft** requires a storage interface with the following functions:
+
+- **Write**:
+
+  ```rust
+  fn write(path: &str, buf: &[u8], put_if_absent: bool) -> Result<bool, io::Error>
```

Comment:
This assumes all the payload must be able to hold in a buffer. Is it desired?

In OpenDAL, the read/write APIs are symmetrical:

* read(path) -> Result<Buffer>
* reader(path) -> Result<Reader>
* write(path, buf, [put_if_absent]) -> Result<()>
* writer(path, [put_if_absent]) -> Result<Writer>

Reply:
Yes. the first version is a prototype with simpler API